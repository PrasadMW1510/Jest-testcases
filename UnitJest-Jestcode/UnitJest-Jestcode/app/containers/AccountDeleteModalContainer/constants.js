/*
 *
 * AccountDeleteModalContainer constants
 *
 */

export const POST_ACCOUNT_DELETE_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST';
export const POST_ACCOUNT_DELETE_REQUEST_FAILURE =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST_FAILURE';
export const POST_ACCOUNT_DELETE_REQUEST_SUCCESS =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST_SUCCESS';
export const POST_ACCOUNT_DELETE_MIA_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_MIA_DELETE_REQUEST';
export const POST_ACCOUNT_UNENROLL_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST';
export const POST_ACCOUNT_UNENROLL_REQUEST_FAILURE =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST_FAILURE';
export const POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS';
