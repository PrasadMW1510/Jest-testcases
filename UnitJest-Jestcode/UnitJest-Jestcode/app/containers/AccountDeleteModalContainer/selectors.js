import { createSelector } from 'reselect';

/**
 * Direct selector to the accountDeleteModalContainer state domain
 */
const selectAccountDeleteModalContainerDomain = state => state.get('accountDeleteModalContainer');

/**
 * Default selector used by AccountDeleteModalContainer
 */

const makeSelectAccountDeleteModalContainer = () =>
  createSelector(selectAccountDeleteModalContainerDomain, substate => substate.toJS());

export default makeSelectAccountDeleteModalContainer;
export { selectAccountDeleteModalContainerDomain };
