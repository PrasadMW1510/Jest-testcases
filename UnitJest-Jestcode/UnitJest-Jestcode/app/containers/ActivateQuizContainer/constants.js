/*
 *
 * ActivateQuizContainer constants
 *
 */

export const ACTIVATE_QUIZ_LIST = 'app/ActivateQuizContainer/activate_quiz_list';
export const DEFAULT_ACTION = 'app/ActivateQuizContainer/DEFAULT_ACTION';
export const ACTIVATE_QUIZ_LIST_FAILURE = 'app/activateQuizContainer/activate-quiz-list-failure';
export const INITIALIZE_MODAL_FLAG = 'app/initialize-modal-flag';
