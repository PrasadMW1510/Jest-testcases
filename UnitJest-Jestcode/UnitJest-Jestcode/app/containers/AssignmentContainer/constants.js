/*
 *
 * AssignmentContainer constants
 *
 */

export const GET_PORTFOLIO_ASSIGNMENT_CLASSES =
  'app/AssignmentContainer/GET_PORTFOLIO_ASSIGNMENT_CLASSES';
export const SET_PORTFOLIO_ASSIGNMENT_CLASSES =
  'app/AssignmentContainer/SET_PORTFOLIO_ASSIGNMENT_CLASSES';
export const GET_PORTFOLIO_ASSIGNMENT_CLASSES_ASS_LIST =
  'app/AssignmentContainer/GET_PORTFOLIO_ASSIGNMENT_CLASSES_ASS_LIST';
export const SET_PORTFOLIO_ASSIGNMENT_CLASSES_ASS_LIST =
  'app/AssignmentContainer/SET_PORTFOLIO_ASSIGNMENT_CLASSES_ASS_LIST';
export const GET_PORTFOLIO_ASSIGNMENT_CLASSES_ASS_LIST_TEMP =
  'app/AssignmentContainer/GET_PORTFOLIO_ASSIGNMENT_CLASSES_ASS_LIST_TEMP';
