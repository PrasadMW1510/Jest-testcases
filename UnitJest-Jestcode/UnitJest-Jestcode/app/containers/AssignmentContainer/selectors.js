import { createSelector } from 'reselect';

/**
 * Direct selector to the assignmentContainer state domain
 */
const selectAssignmentContainerDomain = state => state.get('assignmentContainer');

/**
 * Other specific selectors
 */

/**
 * Default selector used by AssignmentContainer
 */

const makeSelectAssignmentContainer = () =>
  createSelector(selectAssignmentContainerDomain, substate => substate.toJS());

export default makeSelectAssignmentContainer;
export { selectAssignmentContainerDomain };
