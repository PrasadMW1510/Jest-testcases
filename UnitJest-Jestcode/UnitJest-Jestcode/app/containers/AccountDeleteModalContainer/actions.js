/*
 *
 * AccountDeleteModalContainer actions
 *
 */
import { generalFailure } from 'containers/App/actions';
import * as Constants from './constants';

/**
 * Handles API logic for account delete action
 *
 * @param accountDeletePayload
 * @returns {{type}}
 */
export function postAccountDeleteRequest(accountDeletePayload) {
  // payload contains (1)list of accts to delete, (2) searchOpts used to refresh
  return {
    type: Constants.POST_ACCOUNT_DELETE_REQUEST,
    payload: accountDeletePayload,
  };
}

/**
 * Handles API logic for account delete action
 *
 * @param accountDeletePayload
 * @returns {{type}}
 */
export function postAccountDeleteMIARequest(accountDeletePayload) {
  // payload contains (1)list of accts to delete, (2) searchOpts used to refresh
  return {
    type: Constants.POST_ACCOUNT_DELETE_MIA_REQUEST,
    payload: accountDeletePayload,
  };
}

/**
 * Handles API logic for account delete is a success
 *
 *  @returns {{type}}
 */
export function postAccountDeleteRequestSuccess() {
  return {
    type: Constants.POST_ACCOUNT_DELETE_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for when account delete is a failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function postAccountDeleteRequestFailure(error) {
  return generalFailure({
    type: Constants.POST_ACCOUNT_DELETE_REQUEST_FAILURE,
    error,
  });
}

/**
 * Handles API logic for account unenroll action
 *
 * @param accountUnenrollPayload
 * @returns {{type}}
 */
export function postAccountUnenrollRequest(accountUnenrollPayload) {
  // payload contains (1)list of accts to delete, (2) searchOpts used to refresh
  return {
    type: Constants.POST_ACCOUNT_UNENROLL_REQUEST,
    payload: accountUnenrollPayload,
  };
}

/**
 * Handles API logic for account unenroll is a success
 *
 *  @returns {{type}}
 */
export function postAccountUnenrollRequestSuccess() {
  return {
    type: Constants.POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for when account unenroll is a failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function postAccountUnenrollRequestFailure(error) {
  return generalFailure({
    type: Constants.POST_ACCOUNT_UNENROLL_REQUEST_FAILURE,
    error,
  });
}
