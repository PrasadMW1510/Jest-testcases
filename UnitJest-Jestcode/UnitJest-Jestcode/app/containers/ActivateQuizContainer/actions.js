/*
 *
 * ActivateQuizContainer actions
 *
 */

import * as Constants from './constants';

export function activateQuizRequest(activateQuizPreviewData, searchOpts, searchDataChecked) {
  return {
    type: Constants.ACTIVATE_QUIZ_LIST,
    activateQuizPreviewData,
    searchOpts,
    searchDataChecked,
  };
}

export function getPostActivateRequestFailure(error) {
  return {
    type: Constants.ACTIVATE_QUIZ_LIST_FAILURE,
    error,
  };
}

export function defaultAction() {
  return {
    type: Constants.DEFAULT_ACTION,
  };
}

export function initialiseModalFlag() {
  return {
    type: Constants.INITIALIZE_MODAL_FLAG,
  };
}
