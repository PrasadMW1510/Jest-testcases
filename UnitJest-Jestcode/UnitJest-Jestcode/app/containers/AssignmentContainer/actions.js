/*
 *
 * AssignmentContainer actions
 *
 */

import * as Constants from './constants';

export function getClassesDataRequest() {
  return {
    type: Constants.GET_PORTFOLIO_ASSIGNMENT_CLASSES,
  };
}

export function setClassRequestSuccess(data, node) {
  const updTeacherObj = [];
  if (data && data.length > 0) {
    data.map(item => {
      const newItem = item;
      newItem.type = 'PfAsignment';
      newItem.data = [];
      newItem.data.push(node);
      updTeacherObj.push(item);
      return item;
    });
  }
  return {
    type: Constants.SET_PORTFOLIO_ASSIGNMENT_CLASSES,
    updTeacherObj,
  };
}

export function setInboxDataByCommunityId(data, id) {
  const newData = [];
  if (data.length > 0) {
    data.map(item => {
      if (id.indexOf(item.community_id) !== -1 && item.graded === 'false') {
        newData.push(item);
      }
      return item;
    });
  }
  return {
    type: Constants.SET_PORTFOLIO_ASSIGNMENT_CLASSES_ASS_LIST,
    newData,
  };
}

export function getClassAssignmentRequest(data, node, communityId) {
  const newGrid = [];
  if (node.length > 0) {
    node[0].map(item => {
      if (item.classId === data && item.createdForClass === data) {
        newGrid.push(item);
      }
      return item;
    });
  }
  return {
    type: Constants.GET_PORTFOLIO_ASSIGNMENT_CLASSES_ASS_LIST_TEMP,
    newGrid,
    communityId,
    data,
  };
}

export function setClassGridRequestSuccess(data) {
  const newData = data;
  return {
    type: Constants.SET_PORTFOLIO_ASSIGNMENT_CLASSES_ASS_LIST,
    newData,
  };
}
