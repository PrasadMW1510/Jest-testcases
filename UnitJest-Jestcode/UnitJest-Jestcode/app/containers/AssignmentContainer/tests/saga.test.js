import { takeLatest, put, all } from 'redux-saga/effects';
import defaultSaga, * as Saga from '../saga';
import * as Selectors from '../../App/selectors';
import * as Actions from '../actions';
import * as Constants from '../constants';
/* eslint-disable redux-saga/yield-effects */
describe('TeacherMadeQuizContainer Saga', () => {
  let generator = null;
  let loginDataSelector = null;
  let profileSessionIdSelector = null;
  let profileUserIdSelector = null;
  beforeEach(() => {
    profileSessionIdSelector = jest.fn();
    profileUserIdSelector = jest.fn();
    loginDataSelector = jest.fn();
    jest.spyOn(Selectors, 'makeSelectLoginData').mockReturnValue(loginDataSelector);
    jest.spyOn(Selectors, 'makeSelectProfileSessionId').mockReturnValue(profileSessionIdSelector);
    jest.spyOn(Selectors, 'makeSelectProfileUserId').mockReturnValue(profileUserIdSelector);
  });
  // describe('getPfAssignClassDataRequest', () => {
  //   beforeEach(() => {
  //     generator = Saga.getPfAssignClassDataRequest();
  //   });
  //   it('setClassRequestSuccess', () => {
  //     const ServiceResponse = {
  //       output_data: [
  //         {
  //           classes: [
  //             {
  //               class: [
  //                 {
  //                   display_name: ['name1', 'name2'],
  //                   class_id: ['class1'],
  //                 },
  //               ],
  //             },
  //           ],
  //         },
  //       ],
  //     };
  //     const response = [
  //       {
  //         id: 'class1',
  //         type: 'PfAsignment',
  //         name: 'name1',
  //       },
  //     ];
  //     expect(generator.next().value).toMatchSnapshot();
  //     expect(generator.next('SessionIDVal').value).toMatchSnapshot();
  //     expect(generator.next('UserIDVal').value).toMatchSnapshot();
  //     const putDescriptor1 = generator.next(ServiceResponse).value;
  //     expect(putDescriptor1).toEqual(put(Actions.setClassRequestSuccess(response)));
  //   });
  // });
  describe('getPfAssignClassGridDataRequest', () => {
    const classId = {};
    beforeEach(() => {
      generator = Saga.getPfAssignClassGridDataRequest(classId);
    });
    it('setClassRequestSuccess', () => {
      const ServiceResponse = {
        output_data: [
          {
            workItemsMetadata: [
              {
                workItemMetadata: [
                  {
                    $: {
                      read: 'false',
                      graded: false,
                      workItemId: '12',
                      dateSubmitted: '10/01/2018',
                      classId: 'classID',
                      assignment: 'assign',
                      from: 'from',
                      link: ' ',
                      communityId: 'commId',
                      createdForClass: 'createdClass',
                    },
                  },
                ],
              },
            ],
          },
        ],
      };
      const response = [
        {
          id: 'classID',
          date: '10/01/2018',
          classId: 'classID',
          assignment: 'assign',
          from: 'from',
          community_id: 'commId',
          graded: false,
          link: ' ',
          student: undefined,
          createdForClass: 'createdClass',
          workItemId: '12',
        },
      ];
      expect(generator.next().value).toMatchSnapshot();
      expect(generator.next('SessionIDVal').value).toMatchSnapshot();
      expect(generator.next('UserIDVal').value).toMatchSnapshot();
      const putDescriptor1 = generator.next(ServiceResponse).value;
      expect(putDescriptor1).toEqual(put(Actions.setClassGridRequestSuccess(response)));
    });
    it('setClassRequestSuccess', () => {
      const ServiceResponse = {
        output_data: [
          {
            workItemsMetadata: [
              {
                workItemMetadata: undefined,
              },
            ],
          },
        ],
      };
      expect(generator.next().value).toMatchSnapshot();
      expect(generator.next('SessionIDVal').value).toMatchSnapshot();
      expect(generator.next('UserIDVal').value).toMatchSnapshot();
      const putDescriptor1 = generator.next(ServiceResponse).value;
      expect(putDescriptor1).toEqual(put(Actions.setClassGridRequestSuccess([])));
    });
    it('setClassRequestSuccess err part', () => {
      const err = 'mockerror';
      expect(generator.next().value).toMatchSnapshot();
      expect(generator.next('SessionIDVal').value).toMatchSnapshot();
      expect(generator.next('UserIDVal').value).toMatchSnapshot();
      expect(generator.throw(err).value).toMatchSnapshot();
    });
  });
  // describe('saveAssignmentRequest', () => {
  //   beforeEach(() => {
  //     generator = Saga.saveAssignmentRequest();
  //   });
  //   it('setClassRequestSuccess', () => {
  //     const ServiceResponse = {
  //       output_data: [
  //         {
  //           classes: [
  //             {
  //               class: [
  //                 {
  //                   display_name: ['name1', 'name2'],
  //                   class_id: ['class1'],
  //                 },
  //               ],
  //             },
  //           ],
  //         },
  //       ],
  //     };
  //     expect(generator.next().value).toMatchSnapshot();
  //     expect(generator.next('SessionIDVal').value).toMatchSnapshot();
  //     expect(generator.next(ServiceResponse).value).toMatchSnapshot();
  //   });
  // });
  describe('defaultSaga', () => {
    beforeAll(() => {
      generator = defaultSaga();
    });
    it('All calls are made', () => {
      expect(generator.next().value).toEqual(
        all([
          // takeLatest(Constants.GET_PORTFOLIO_ASSIGNMENT_CLASSES, Saga.getPfAssignClassDataRequest),
          takeLatest(
            Constants.GET_PORTFOLIO_ASSIGNMENT_CLASSES_ASS_LIST,
            Saga.getPfAssignClassGridDataRequest
          ),
        ])
      );
    });
  });
});
