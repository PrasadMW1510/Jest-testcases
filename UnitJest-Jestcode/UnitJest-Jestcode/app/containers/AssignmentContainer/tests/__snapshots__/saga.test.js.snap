// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherMadeQuizContainer Saga getPfAssignClassGridDataRequest setClassRequestSuccess 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`TeacherMadeQuizContainer Saga getPfAssignClassGridDataRequest setClassRequestSuccess 2`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`TeacherMadeQuizContainer Saga getPfAssignClassGridDataRequest setClassRequestSuccess 3`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "SessionIDVal",
      "UserIDVal",
      Object {},
      "<>",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`TeacherMadeQuizContainer Saga getPfAssignClassGridDataRequest setClassRequestSuccess 4`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`TeacherMadeQuizContainer Saga getPfAssignClassGridDataRequest setClassRequestSuccess 5`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`TeacherMadeQuizContainer Saga getPfAssignClassGridDataRequest setClassRequestSuccess 6`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "SessionIDVal",
      "UserIDVal",
      Object {},
      "<>",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`TeacherMadeQuizContainer Saga getPfAssignClassGridDataRequest setClassRequestSuccess err part 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`TeacherMadeQuizContainer Saga getPfAssignClassGridDataRequest setClassRequestSuccess err part 2`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`TeacherMadeQuizContainer Saga getPfAssignClassGridDataRequest setClassRequestSuccess err part 3`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "SessionIDVal",
      "UserIDVal",
      Object {},
      "<>",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`TeacherMadeQuizContainer Saga getPfAssignClassGridDataRequest setClassRequestSuccess err part 4`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "newData": Array [],
      "type": "app/AssignmentContainer/SET_PORTFOLIO_ASSIGNMENT_CLASSES_ASS_LIST",
    },
    "channel": null,
  },
}
`;
